starport/services/network/networktypes/genesisinformation.go

```diff
@@ -77,59 +77,136 @@ func NewGenesisInformation(
 func (gi GenesisInformation) GetVestingAccounts() (accs []VestingAccount) {
-	for _, vestingAcc := range gi.VestingAccounts {
-		accs = append(accs, vestingAcc)
-	}
+	accs = make([]VestingAccount, len(gi.VestingAccounts))
+	copy(accs, gi.VestingAccounts)
 	return accs
 }
 
 // GetGenesisValidators converts into array and returns genesis validators
 func (gi GenesisInformation) GetGenesisValidators() (vals []GenesisValidator) {
-	for _, genVal := range gi.GenesisValidators {
-		vals = append(vals, genVal)
-	}
+	vals = make([]GenesisValidator, len(gi.GenesisValidators))
+	copy(vals, gi.GenesisValidators)
 	return vals
 }
 
+func (gi GenesisInformation) ContainsGenesisAccount(address string) bool {
+	for _, account := range gi.GenesisAccounts {
+		if account.Address == address {
+			return true
+		}
+	}
+	return false
+}
+func (gi GenesisInformation) ContainsVestingAccount(address string) bool {
+	for _, account := range gi.VestingAccounts {
+		if account.Address == address {
+			return true
+		}
+	}
+	return false
+}
+func (gi GenesisInformation) ContainsGenesisValidator(address string) bool {
+	for _, account := range gi.GenesisValidators {
+		if account.Address == address {
+			return true
+		}
+	}
+	return false
+}
+
+func (gi *GenesisInformation) AddGenesisAccount(acc GenesisAccount) {
+	gi.GenesisAccounts = append(gi.GenesisAccounts, acc)
+}
+
+func (gi *GenesisInformation) AddVestingAccount(acc VestingAccount) {
+	gi.VestingAccounts = append(gi.VestingAccounts, acc)
+}
+
+func (gi *GenesisInformation) AddGenesisValidator(val GenesisValidator) {
+	gi.GenesisValidators = append(gi.GenesisValidators, val)
+}
+
+func (gi *GenesisInformation) RemoveGenesisAccount(address string) {
+	for i, account := range gi.GenesisAccounts {
+		if account.Address == address {
+			gi.GenesisAccounts = append(gi.GenesisAccounts[:i], gi.GenesisAccounts[i+1:]...)
+		}
+	}
+}
+
+func (gi *GenesisInformation) RemoveVestingAccount(address string) {
+	for i, account := range gi.VestingAccounts {
+		if account.Address == address {
+			gi.VestingAccounts = append(gi.VestingAccounts[:i], gi.VestingAccounts[i+1:]...)
+		}
+	}
+}
+
+func (gi *GenesisInformation) RemoveGenesisValidator(address string) {
+	for i, account := range gi.GenesisValidators {
+		if account.Address == address {
+			gi.GenesisValidators = append(gi.GenesisValidators[:i], gi.GenesisValidators[i+1:]...)
+		}
+	}
+}
+
+func (gi GenesisInformation) GetGenesisAccount(address string) GenesisAccount {
+	for _, account := range gi.GenesisAccounts {
+		if account.Address == address {
+			return account
+		}
+	}
+	return GenesisAccount{}
+}
+
+func (gi GenesisInformation) GetVestingAccount(address string) VestingAccount {
+	for _, account := range gi.VestingAccounts {
+		if account.Address == address {
+			return account
+		}
+	}
+	return VestingAccount{}
+}
+
+func (gi GenesisInformation) GetGenesisValidator(address string) GenesisValidator {
```

Comment:
Can't see where this func is being used at? I'm checking in the PR and couldn't find a place except the test. Can we remove funcs that are not used, we don't need to introduce funcs just for tests.